### PR TITLE
Add Pandas for Testing

### PR DESCRIPTION
**Pull request by rymarczy**
Pandas added to project as Poetry dev dependency.

Additional file detail checks added to `test_convert.py`
- numpy dataframe shape
- nan value count for each data column
- dtype check for each data column
- min value check for each data column (if possible)
- max value check for each data column (if possible)

### CODE REVIEW

File: py_gtfs_rt_ingestion/pyproject.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ boto3 = "^1.23.3"
 
 [tool.poetry.dev-dependencies]
 pytest = "^5.2"
+pandas = "^1.4.2"
```
Comment (rymarczy) [inline py_gtfs_rt_ingestion/pyproject.toml:15]: Add pandas as dev dependency in poetry, I think pyarrow already includes pandas, but better to be explicit.

File: .gitignore [hunk 2]
```diff
@@ -11,3 +11,5 @@ erl_crash.dump
 
 
 __pycache__
+venv
+*.ipynb
```
Comment (rymarczy) [inline .gitignore:15]: helpful additions to .gitignore for me ;)

File: py_gtfs_rt_ingestion/tests/test_convert.py [hunk 3]
```diff
@@ -15,52 +15,54 @@ def test_vehicle_positions_file_conversion(tmpdir):
...
+        'direction_id':(4,'float64','1.0','0.0'),
+        'route_id':(0,'object','Shuttle-Generic','1'),
+        'schedule_relationship':(0,'object','UNSCHEDULED','ADDED'),
+        'start_date':(40,'object','nan','nan'),
+        'start_time':(87,'object','nan','nan'),
+        'trip_id':(426,'object','nan','nan'),
+        'vehicle_id':(0,'object','y4124','1625'),
+        'vehicle_label':(0,'object','4124','0420'),
+        'vehicle_consist':(324,'object','nan','nan'),
+    }
```
Comment (rymarczy) [inline py_gtfs_rt_ingestion/tests/test_convert.py:47]: There's definitely a better way to do this, but this dictionary has each Table column as a key and then count of (na values in rows, dtype, min and max) values stored in a tuple. 

Not all columns can have min/max values, those have 'nan' stored for min/max.

File: py_gtfs_rt_ingestion/tests/test_convert.py [hunk 4]
```diff
@@ -15,52 +15,54 @@ def test_vehicle_positions_file_conversion(tmpdir):
...
+
+    assert np_df.shape == (426,25)
```
Comment (rymarczy) [inline py_gtfs_rt_ingestion/tests/test_convert.py:49]: compare entire shape of data frame to expected (rows,cols)

File: py_gtfs_rt_ingestion/tests/test_convert.py [hunk 5]
```diff
@@ -15,52 +15,54 @@ def test_vehicle_positions_file_conversion(tmpdir):
...
+
+    all_expected_paths = set(file_details.keys())
 
     # ensure all of the expected paths were found and there aren't any
     # additional ones
-    assert all_expected_paths == found_paths
-
-    # i counted 426 in the input json
-    assert table.num_rows == 426
+    assert all_expected_paths == set(np_df.columns)
```
Comment (rymarczy) [inline py_gtfs_rt_ingestion/tests/test_convert.py:55]: streamline expected paths checking

File: py_gtfs_rt_ingestion/tests/test_convert.py [hunk 6]
```diff
@@ -15,52 +15,54 @@ def test_vehicle_positions_file_conversion(tmpdir):
...
+
+    # check file details
+    for col, (na_count, d_type, max, min) in file_details.items():
+        print(f"checking: {col}")
+        assert na_count == np_df[col].isna().sum()
+        assert d_type == np_df[col].dtype
+        if max != 'nan':
+            assert max == str(np_df[col].max())
+        if min != 'nan':
+            assert min == str(np_df[col].min())
```
Comment (rymarczy) [inline py_gtfs_rt_ingestion/tests/test_convert.py:65]: Loop to check (na_count, d_type, max, min) aggregations for each column. 

max/min only check if not saved as 'nan' in tuple values.

**Comment by mzappitello** [comment]
LGTM.

Merging so that I can put mine in.

**Pull request merged**